### PR TITLE
Set employee relationship to inactive if contact is a child

### PR DESCRIPTION
**Pull request by Edzelopez**

### CODE REVIEW

File: aoservicelisting.php [hunk 1]
```diff
@@ -160,6 +160,28 @@ function aoservicelisting_civicrm_pre($op, $objectName, $id, &$params) {
...
+
+    if ($params['relationship_type_id'] == 5) {
+      $contact = civicrm_api3('Contact', 'get', [
+        'sequential' => 1,
+        'return' => ["contact_sub_type"],
+        'id' => $params['contact_id_a'],
+      ]);
+      if (!empty($contact['values']) && in_array('Child', $contact['values'][0]['contact_sub_type'])) {
+        // This is a child contact, set the relationship that was created as inactive.
+        $params['is_active'] = 0;
```
Comment (seamuslee001) [inline aoservicelisting.php:170]: should we also set an End Date as well? @Edzelopez

**Comment by seamuslee001** [comment]
@Edzelopez I have tested this out on jma.staging and it seems that this prevents the Edit Relationship form form submitting at all. If you checkout master then the form submits with these changes it doesn't I'm wondering if the API calls are creating some kind of infinite loop

**Review by seamuslee001 — APPROVED** [review]

**Comment by seamuslee001** [comment]
I have tested this on dev and confirmed that editing relationships works as expected and that other employer of relationships are not affected if they are to an authorised contact or any other contact that is not a Child. I have also updated the service listing view configuration as well just in case to mention that relationship needs to be active

**Comment by JoeMurray** [comment]
So it feels inelegant to leave an employee relationship that in real world
never existed. I imagine there are technical reasons for not a) deleting
these and b) preventing their creation.

**Comment by seamuslee001** [comment]
This has been deployed to production merging

**Pull request merged**